api/endpoint.go

```diff
@@ -122,110 +122,64 @@ func (e *Endpoint) makeDataRequest(
 	writeResponse(ctx, metadata, data)
 }
 
-func (e *Endpoint) slice(ctx *gin.Context, request SliceRequest) {
-	prepareRequestLogging(ctx, request)
-	conn, err := e.MakeVdsConnection(request.Vds, request.Sas)
-	if abortOnError(ctx, err) {
-		return
-	}
-
-	cacheKey, err := request.hash()
-	if abortOnError(ctx, err) {
-		return
-	}
-
-	cacheEntry, hit := e.Cache.Get(cacheKey)
-	if hit && conn.IsAuthorizedToRead() {
-		ctx.Set("cache-hit", true)
-		writeResponse(ctx, cacheEntry.Metadata(), cacheEntry.Data())
-		return
-	}
-
-	handle, err := core.NewVDSHandle(conn)
-	if abortOnError(ctx, err) {
-		return
-	}
-	defer handle.Close()
-
+func (request SliceRequest) execute(
+	handle core.VDSHandle,
+) (data [][]byte, metadata []byte, err error) {
 	axis, err := core.GetAxis(strings.ToLower(request.Direction))
-	if abortOnError(ctx, err) {
+	if err != nil {
 		return
```

Comment:
I am not completely sure but just to confirm, Will it return a proper error message in case of an error?

Is it not supposed to be like this 🤔

`return nil, nil, err `

Reply:
I originally had it like
`return nil, nil, err `

But there are so many returns here, that it just becomes annoying.
So I decided to use go feature [Named return values](https://go.dev/tour/basics/7)

Wouldn't use it in simple functions, as in my mind they harm readability, but here the number of returns warrants it. 
It also has some additional consequences. Like if there error is in data function, it will return `nil, metadata, error` instead of nil, nil, error. But it should be ok as far as I can see.
Also if err is not nil it will return whatever err is, so we should be safe.

Haven't use this feature before, so we can see if it was a mistake to use it long-term?